src/axiosWrappers.ts

```diff
@@ -44,15 +45,18 @@ export function listenForNonSuccessStatusCode(
 }
 
 /** @hidden */
-export function rawRequest<ResponseType>(
+export async function rawRequest<ResponseType>(
   axiosInstance: AxiosInstance,
   requestConfig: AxiosRequestConfig,
   isSafeToRetry: boolean = false
 ) {
   if (isSafeToRetry) {
     makeRequestSafeToRetry(requestConfig);
   }
-  return axiosInstance(requestConfig).catch(handleErrorResponse) as Promise<
-    AxiosResponse<ResponseType>
-  >;
+  requestConfig.data = transformDateInRequest(requestConfig.data);
+  const response = (await axiosInstance(requestConfig).catch(
+    handleErrorResponse
```

Comment:
what I don't like here is that you mix `.then(), .catch()` syntax and `await, try{}` for promises. I mean, you take a style from both of them and it's a bit confusing to my mind.
I've used to use either one or another. But let's talk about this more.

Reply:
I will clean up